### DISCUSSION TITLE
PHP 8 would not start via services

### DISCUSSION THREAD

**Discussion by dmetzler1988** [Homebrew/discussions#217] [Everyday usage] [answered]
Hi,

after brew update && brew upgrade thx new PHP 8.0.0 was installed. But when i start it via `brew services start php` it have an error status.
I can't find any logs which told me something about this error.

Does anybody have/had this error also? And does anyone know how to fix this?
I can't start the PHP 8 container - the others (e.g. PHP 7.4) works fine.

I also deinstalled the php (with brew cleanup and brew doctor) and reinstalled it again (alltimes without sudo).

**Comment by SMillerDev** [comment, marked as answer]
What happens when you run `php-fpm`?

The logs should be in `$HOMEBREW_PREFIX/var/log/php-fpm.log`

**Reply by dmetzler1988** [reply]
It returns the following lines:

```
[03-Dec-2020 15:53:20] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[03-Dec-2020 15:53:20] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[03-Dec-2020 15:53:20] ERROR: unable to bind listening socket for address '127.0.0.1:9980': Address already in use (48)
[03-Dec-2020 15:53:20] ERROR: FPM initialization failed
```

I checked the port with `lsof -i -P | grep -i 9980` and the result here is:

```
php-fpm   2323 metz    8u  IPv4 0xbadd876cef9b304b      0t0  TCP localhost:9980 (LISTEN)
php-fpm   2339 metz    9u  IPv4 0xbadd876cef9b304b      0t0  TCP localhost:9980 (LISTEN)
php-fpm   2340 metz    9u  IPv4 0xbadd876cef9b304b      0t0  TCP localhost:9980 (LISTEN)
```

I changed the port to another one, which was previously not set - but with same result (on the different port).

The output in the `php-fpm.log` doesn't tell more:

```
...
[03-Dec-2020 16:00:52] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[03-Dec-2020 16:00:52] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[03-Dec-2020 16:00:52] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[03-Dec-2020 16:00:52] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[03-Dec-2020 16:00:52] NOTICE: fpm is running, pid 3802
[03-Dec-2020 16:00:52] NOTICE: ready to handle connections
```

I use MacOS Big Sure - maybe this is also a useful information.

**Reply by SMillerDev** [reply]
Your old PHP was still running and the processes aren't compatible. It should work after you `kill -9 2323`, `kill -9 2339` and `kill -9 2340`.

**Reply by SMillerDev** [reply]
If that doesn't work you should post the full output of your `php-fpm` run.

**Reply by dmetzler1988** [reply]
The three processes will not be there, if i stop the php service - and also not, when i change the port to e.g. 9910 (the port 9980 port is free then).

But i will try to kill it manually and start again.

The first output is the complete output, that php-fpm returns, when i run it.

**Reply by dmetzler1988** [reply]
Ok, i tried to kill the processes - but if i kill it, a few seconds later new processes are up with the same port - and if i run `php-fpm` again - the same result occurs.

I tested the same with my working php@7.4 - php-fpm returns there the same error but with the ports from 7.4 (9974). But 7.4 doesn't have any issues.

**Reply by SMillerDev** [reply]
So it does actually start and work using `brew services start php` then? Because it seems the service just keeps running in the background regardless of what `brew services` tells you. If that's the case I'm not sure what the issue is.

**Reply by dmetzler1988** [reply]
Ok, you are right - it seems that it runs today (but not so yesterday - maybe the new update & upgrade today helped).

So only the `brew services list` output with the `error` state on php is very confusing.

```
php       error   metz /Users/metz/Library/LaunchAgents/homebrew.mxcl.php.plist
php@7.3   started metz /Users/metz/Library/LaunchAgents/homebrew.mxcl.php@7.3.plist
php@7.4   started metz /Users/metz/Library/LaunchAgents/homebrew.mxcl.php@7.4.plist
```
